### ISSUE TITLE
[Hyperopt] Memory pressure with windowing

### ISSUE THREAD

**Issue by ShreyaR** [ludwig-ai/ludwig#2148]
Currently, we do pretty aggressive windowing in order to support large datasets in hyperopt.

For every trial in hyperopt, we set the `window size = <total object store memory> / 40`. However, based on a discussion with Ray Data folks, this should be much smaller at a factor of 2-3.

**Comment by arnavgarg1** [comment]
@ShreyaR Is the goal to dynamically adjust the window size so that each window size is manageable irrespective of the number of windows? If it is the latter (which is what I'm guessing it is), is there a more clear idea of what's considered to be less aggressive/better?

A couple of questions to understand this better:

- Is there a reason we chose to divide by 40 in the first place, and is this something that's been hard coded? If not, do we automatically configure the window size? 
- Are you able to link me to the code block where we set the window size this way?

**Comment by ShreyaR** [comment]
Good questions!

The issue we're seeing is that with the current window size, we shouldn't be experiencing the memory pressure that we currently are. So the goal for solving this issue is to understand where the memory pressure comes from, and if there's anything we can do in how we set up windowing to reduce that.

Memory pressure is exhibited as a full object store and consequent spilling of objects to disk. Based on discussions with the ray team, we shouldn't observe any disk spillage given the current window size of `<total object store memory (tosm)> / 40`. Something less aggressive would a window size of `<tosm> / <num_trials * 2.5>`.

- We divide by 40 because for a single trial, the recommendation is to have a window size of `<tosm> / 5`. Since we run 8 trials for hyperopt, with each trial doing data loading shuffling independently, we set the current window size as `<tosm> / (5 * 8)`. This isn't hardcoded in Ludwig today, and is expected to be provided by the user.
- If we set the window size to something larger like `<tosm> / <num_trials * 2.5> ` and don't see any object spillage, that would be validating that less aggressive windowing works.
